modules/opengl/src/buffer/bufferobjectarray.cpp

```diff
@@ -53,9 +58,9 @@ BufferObjectArray& BufferObjectArray::operator=(const BufferObjectArray& that) {
     if (this != &that) {
         if (id_ != 0) {
             // avoid creating the vertex array if is not already initialized
-            bind();
             clear();
```

Comment:
now this looks wrong, you unbind without a bind... If clear binds should it not also unbind then? this seems messy.

Reply:
Probably `clear()` should not call `glDisableVertexAttribArray` in the first place. Then there is no need for bind/unbind at all. However, one would then need to call `bind()` afterward to disable the vertex attribs that are still enabled, which is also somewhat strange. And in case the BOA is bound before calling clear(), it stays bound along with all attributes despite having called clear(). 
It all comes down to mixing the state machine of OpenGL (bound BOA) and our internal states.